src/composables/usePool.ts

```diff
@@ -419,39 +419,39 @@ export function removeBptFrom(pool: Pool): Pool {
 /**
  * Updates the passed subPool by removing its pre-minted tokens.
  */
-export function removeBptFromTree(tree: SubPool) {
-  if (tree.tokens) {
-    removePremintedToken(tree);
+export function removeBptFromPool(pool: Pool | SubPool) {
+  if (pool.tokens) {
+    removeBptFromTokens(pool);
 
-    tree.tokens.forEach(token => {
+    pool.tokens.forEach(token => {
       if (token.token?.pool) {
-        removeBptFromTree(token.token.pool);
+        removeBptFromPool(token.token.pool);
       }
     });
   }
-  return tree;
+  return pool;
 }
 
 /**
  * Updates the passed subPool by removing the preminted token from tokens and updating mainIndex accordingly.
  */
-function removePremintedToken(tree: SubPool) {
-  if (!tree.tokens) {
+export function removeBptFromTokens(pool: Pool | SubPool) {
```

Comment:
Update function name to be a bit more clear. We don't use full 'preminted' terminology elsewhere.